packages/context/src/provider.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Context} from './context';
+import {BoundValue, ValueOrPromise} from './binding';
+import {Constructor, instantiateClass} from './resolver';
+
+/**
+ * Provider of a value
+ */
+export interface Provider<T> {
+  value(ctx?: Context): ValueOrPromise<T>;
```

Comment:
Why is `ctx` argument optional? As I read this interface, it means that callers of `Provider.value()` can invoke it with no arguments and the function will still return the provided value.  Which does not seem to be the case of the concrete Provider implementations below.